python/ray/experimental/workflow/api.py

```diff
@@ -0,0 +1,34 @@
+import logging
+import time
+import types
+import uuid
+
+import ray
+from ray.experimental.workflow.workflow_mananger import WorkflowStepFunction, Workflow
+from ray.experimental.workflow import workflow_context
+
+logger = logging.getLogger(__name__)
+
+
+def step(func_or_class) -> WorkflowStepFunction:
+    if isinstance(func_or_class, types.FunctionType):
+        return WorkflowStepFunction(func_or_class)
+    raise NotImplementedError
+
+
+def run(entry_workflow: Workflow, workflow_root_dir=None, workflow_id=None) -> ray.ObjectRef:
+    if workflow_id is None:
+        # TODO(suquark): include the name of the workflow in the default ID,
+        # this makes the ID more readable.
+        workflow_id = uuid.uuid4().hex + '.' + str(time.time())
+    logger.info(f"Workflow job {workflow_id} created.")
+    workflow_context.init_workflow_context(workflow_id, workflow_root_dir)
+    output = entry_workflow.execute()
+    if not isinstance(output, (ray.ObjectRef, tuple)):
```

Comment:
In which case will the output be a tuple?

Reply:
I just think of the cases where the workflow step wraps a remote function that returns a tuple of ObjectRefs. Do you think this is possible case in the future? Otherwise I can enforce the output to be ObjectRefs only.

Reply:
I leave a comment around this line. We'll figure out the actual design when we finalize the API.